File: server/scratch/find_transport_models.js
const fs = require('fs');
const path = require('path');

const schemaPath = path.join(__dirname, '..', 'prisma', 'schema.prisma');
const content = fs.readFileSync(schemaPath, 'utf8');

const regex = /model\s+(\w+)\s+\{[\s\S]*?\}/g;
let match;
while ((match = regex.exec(content)) !== null) {
  const modelName = match[1];
  if (modelName.toLowerCase().includes('transport') || 
      modelName.toLowerCase().includes('vehicle') || 
      modelName.toLowerCase().includes('route') || 
      modelName.toLowerCase().includes('stop') ||
      modelName.toLowerCase().includes('allocation')) {
    console.log(`--- MODEL: ${modelName} ---`);
    console.log(match[0]);
    console.log('\n');
  }
}

File: scratch/find_transport_models.dart
import 'dart:io';

void main() {
  final file = File('c:/edusphere/edusphere-erp-app/server/prisma/schema.prisma');
  if (!file.existsSync()) {
    print('Prisma schema file does not exist');
    return;
  }
  
  final content = file.readAsStringSync();
  final regex = RegExp(r'model\s+(\w+)\s+\{[^}]*\}');
  final matches = regex.allMatches(content);
  
  for (final match in matches) {
    final modelName = match.group(1) ?? '';
    final modelText = match.group(0) ?? '';
    final lowerName = modelName.toLowerCase();
    if (lowerName.contains('transport') || 
        lowerName.contains('vehicle') || 
        lowerName.contains('route') || 
        lowerName.contains('stop') || 
        lowerName.contains('allocation')) {
      print('--- MODEL: $modelName ---');
      print(modelText);
      print('\n');
    }
  }
}
